template/tree_v4.go

```diff
@@ -609,12 +609,27 @@ func (t *TreeV4) FindDeepestTag(address patricia.IPv4Address) (bool, GeneratedTy
 // - use FindDeepestTagsAppend if you can reuse slices, to cut down on allocations
 func (t *TreeV4) FindDeepestTags(address patricia.IPv4Address) (bool, []GeneratedType) {
 	ret := make([]GeneratedType, 0)
-	return t.FindDeepestTagsAppend(ret, address)
+	return t.FindDeepestTagsWithFilterAppend(ret, address, nil)
+}
+
+// FindDeepestTagsWithFilter finds all tags at the deepest level in the tree, matching the provided filter, representing the closest match
+// - use FindDeepestTagsWithFilterAppend if you can reuse slices, to cut down on allocations
+// - returns true regardless of the result of the filtering function
+func (t *TreeV4) FindDeepestTagsWithFilter(address patricia.IPv4Address, filterFunc FilterFunc) (bool, []GeneratedType) {
+	ret := make([]GeneratedType, 0)
+	return t.FindDeepestTagsWithFilterAppend(ret, address, filterFunc)
 }
 
 // FindDeepestTagsAppend finds all tags at the deepest level in the tree, representing the closest match
 // - appends results to the input slice
 func (t *TreeV4) FindDeepestTagsAppend(ret []GeneratedType, address patricia.IPv4Address) (bool, []GeneratedType) {
+	return t.FindDeepestTagsWithFilterAppend(ret, address, nil)
+}
+
+// FindDeepestTagsWithFilterAppend finds all tags at the deepest level in the tree, matching the provided filter, representing the closest match
+// - appends results to the input slice
+// - returns true regardless of the result of the filtering function
```

Comment:
this comment (and corresponding callers) is actually slightly misleading because this can return false if the tree is empty. since this is the only scenario that can happen and this review is long overdue, I think it's fine to fix as part of a later commit/fix